### ISSUE TITLE
Question about calculate_logits function

### ISSUE THREAD

**Issue by id4thomas** [dwlmt/Story-Untangling#7]
In Figure 1 of the paper "Modelling Suspense in Short Stories as Uncertainty Reduction over Neural Representation" the logits for discriminatory loss seems to be calculated with only the encoded story embeddings (story_enc).
However, in _calculate_logits_ and _caculate_discriminatory_loss_ functions in "story_untangling/models/uncertain_reader.py" the logit is calculated using _encoded_stories_flat_ (story_enc) and **target_encoded_sentences_flat** which seems to be the output of the sentence encoder seq2vec model (sent_enc) calculated in the _encode_story_vectors_ function.
Shouldn't this be calculated with only encoded_stories_flat vectors?

Thank you in advance

**Comment by dwlmt** [comment]
Thanks for the query, will check this now.

**Comment by dwlmt** [comment]
You are right about this is was right before see this commit - https://github.com/dwlmt/Story-Untangling/commit/c5f13c2114b9c1bfbe7ed589724c3b3dd2dfe77d . It's a long time ago now and I can't remember why this was changed. I may have been testing what happens if you try to run the loss directly against the sentence representation as opposed to the higher-level representation. But it's not consistent with the paper or experiments so I'll revert it now.

**Comment by dwlmt** [comment]
I've made the change now to the truck and branch so it is now consistent with the paper. Can I also ask what you are looking at the paper for? Anything specific you are looking to do with it or build on? I noticed you have also looked at the TRIPOD Turning Points dataset.

**Comment by id4thomas** [comment]
Thank you very much!
I came across your work while I was looking into the TRIPOD dataset for protagonist conflict detection research.
My goal was to first reproduce your results (mainly Figure 2) and look at ways to utilize the story_enc model outputs to calculate story event continuation probabilities P(e_t+1 | e_1,.. e_t+1)

**Comment by dwlmt** [comment]
As you from the instructions the code is pretty messy and hard to run, and tied to the Edinburgh env. You might be better building on your own setup Also Nelly who wrote the TRIPOD paper as follow up work on this looking at multimodal content for the same movies. So this is something you may also be interested in.

**Comment by id4thomas** [comment]
I managed to run the code by modifying most of the environment related parts. I'll make sure to check the follow up paper you mentioned. Thank you so much for your help
